spec/message.abnf

```diff
@@ -0,0 +1,52 @@
+message = [s] *(declaration [s]) body [s]
+
+declaration = "let" s variable [s] "=" [s] "{" [s] expression [s] "}"
+body = pattern
+     / (selectors 1*([s] variant))
+
+pattern = "{" *(text / placeholder) "}"
+selectors = "match" 1*([s] selector)
+selector = "{" [s] expression [s] "}"
+variant = "when" 1*(s key) [s] pattern
+key = nmtoken / literal / "*"
+
+placeholder = "{" [s] (expression / markup / markup-end) [s] "}"
+
+expression = ((literal / variable) [s annotation])
+           / annotation
+annotation = function *(s option)
+option = name [s] "=" [s] (literal / nmtoken / variable)
+
+markup = markup-start *(s option)
```

Comment:
This requires markup options to be k-v pairs, e.g. these are not okay:

> {+markup false}
> {+checkbox checked}

Do we want to be that strict?

Reply:
I'd prefer to start with the exact same option=value syntax as in function calls. Please file an issue if you'd like to suggest making markup different?